src/test/e2e-emulated/contracts/constants.tact

```diff
@@ -10,6 +19,76 @@ struct T {
     s: S;
 }
 
+// Global functions
+
+// Test assignments
+fun globalFunc1(v: Int): Int {
+    let i = v;
+    let j = 1;
+    let k = i + j;
+    k = k * 10;
+    k += 2;
+    k -= 1;
+    k /= 2;
+    k *= 5;
+    k %= 2;
+    k |= 9;
+    k &= 22;
+    k ^= 3;
+    return k;
+}
+
+// Test repeat
+fun globalFunc2(v: Int): Int {
+    let j = v;
+    repeat(10) {
+        let i = 1;
+        j += i;
+    }
+    return j;
+}
+
+// Test do..until
+fun globalFunc3(v: Int): Int {
+    let i = 20;
+    do {
+        i -= 1;
+    } until (i <= v);
+    return i;
+}
+
+// Test while
+fun globalFunc4(v: Int): Int {
+    let i = 20;
+    while (i <= v) {
+        i += 1;
+    }
+    return i;
+}
+
+// MAX, test conditionals
+fun globalFunc5(a: Int, b: Int): Int {
+    if (a <= b) {
+        return b;
+    } else {
+        return a;
+    }
+}
+
+// GCD for non-negative integers
+// test recursion
```

Comment:
let's also add the factorial function :)
would be nice to have Fibonacci as well, but the definitions are topologically sorted before codegen, so it will error out

Reply:
I did not understand why Fibonacci would fail.

Reply:
Ah, sorry Fibonacci work, I was thinking about mutually recursive even/odd functions but my fingers typed Fibonacci :)